File: src/main/java/pl/laskowski/marcin/model/dividers/ColumnDivider.java
package pl.laskowski.marcin.model.dividers;

import pl.laskowski.marcin.model.Field;
import pl.laskowski.marcin.model.Region;
import pl.laskowski.marcin.model.Sudoku;

import java.util.*;

/**
 * Created by Marcin Laskowski.
 */

public class ColumnDivider implements SudokuDivider {

    @Override
    public Set<Region> divideIntoRegions(Sudoku sudoku) {
        Set<Region> set = new HashSet<>();
        for (int x = 0; x < sudoku.sizeX(); x++){
            Set<Field> fields = new HashSet<>();
            for (int y = 0; y < sudoku.sizeY(); y++) {
                fields.add(sudoku.at(x, y));
            }
            set.add(new Region(fields));
        }
        return set;
    }

}

File: app/src/main/java/pl/laskowski/marcin/model/dividers/ColumnDivider.kt
package pl.laskowski.marcin.model.dividers

import pl.laskowski.marcin.model.Board
import pl.laskowski.marcin.model.Field
import pl.laskowski.marcin.model.Region
import java.util.HashSet

class ColumnDivider : SudokuDivider {

    override fun divideIntoRegions(board: Board): Set<Region> {
        val set: MutableSet<Region> = HashSet()
        for (x in 0 until board.sizeX()) {
            val fields: MutableSet<Field> = HashSet()
            for (y in 0 until board.sizeY()) {
                fields.add(board.at(x, y)!!)
            }
            set.add(Region(fields))
        }
        return set
    }
}